design/controlling-agent-updates.md

```diff
@@ -0,0 +1,66 @@
+# Controlling agent updates
+
+The agent is considered an extension of the service.
+As such, we have an auto-update mechanism to ensure that new features can rely on the agent support they require.
+Conceptually, it's no different from binaries on the app tier or JavaScript served to the web frontend.
+
+For instance, tasks express a minimum agent version required.
+If a selected agent doesn't meet that requirement, the agent is first asked to update itself.
+Other ways an agent can be updated include using a feature which demands an evergreen agent (YAML pipelines, for instance) or clicking the "upgrade all agents" button in a pool.
+
+For some highly-regulated customers, this kind of evergreen infrastructure is at odds with their compliance needs.
+Such customers today are doing things like ACLing the agent's installation directory so it can't update itself or recompiling the agent without update support.
+This leaves us in a situation where no one is happy:
+we can't service the product correctly, and the customer has to maintain custom configuration / infrastructure to meet their compliance needs.
+
+## Solution
+
+_See below for other solutions considered._
+
+We will add a mode to the project-level pool which puts it into "no update" mode.
```

Comment:
Could we simply have a flag on the agent config that says “AutoUpdate=false”? That attribute could be sent to the server and the server could then evaluate the agents based on this flag. This might satisfy the customers that "don't trust the server". Agents with AutoUpdate turned off will simply reject any calls to update themselves. We could also clearly show this flag in the pool/queue UI.

Reply:
I've added a section in the "rejected approaches" area. But we should talk about this after the holiday, it's possible I'm wrong :grin: